lib/filter-schema.js

```diff
@@ -12,9 +12,9 @@ function filterIntrospectionSchema (app, policy, { applyPolicy: policyFunction }
 
     // the filter hook must be the last one to be executed (after all the authContextHook ones)
     app.ready(err => {
-      /* istanbul ignore next */
-      if (err) throw err
-      app.graphql.addHook('preExecution', filterGraphQLSchemaHook.bind(app))
+      if (!err) {
+        app.graphql.addHook('preExecution', filterGraphQLSchemaHook.bind(app))
+      }
```

Comment:
Are you sure about this? Isn't it swallowing the error?

Reply:
Yeah I thought the same and it's related to the issue I was coming up against! If I leave as it, `app.ready()` rejects with the expected error but then the error is thrown again within this callback and causes unhandled rejection issues.

Even though we're swallowing the error here, it's still being thrown, as shown and caught in this test here: https://github.com/mercurius-js/auth/blob/main/test/introspection-filter.js#L359

wdyt? Reckon it worth looking for a better solution?

Reply:
Hey @mcollina ! Hope the Fastify release went okay this week! Just wondered what your thoughts were on my response to your comment? :)

Reply:
I've actually refactored it to use the fastify `onReady` hook instead, lemme know what you think!

Reply:
Yep, it should be okay from what I've read and the tests cover it and all run absolutely fine: https://github.com/mercurius-js/auth/blob/main/test/introspection-filter.js#L104

The only things I can think of with changing to use the `onReady` hook:
* It does fire slightly earlier that waiting until the `app.ready` callback to fire but crucially, it still ensures that it fires the filter schema hook after **all** the auth context hooks are called.
* If another `onReady` hook was registered after the plugin registration, this this would run after (rather than before with the original implementation) but I would argue that in this case, this introduces dependencies and you could get around it by registering the `onReady` before the plugin registration

wdyt?

cc. @Eomm for visibility as I think you did the original implementation of this :)

Reply:
Need to check - I don't remember ATM